docs/adrs/2023-05-08-exposing-metrics.md

```diff
@@ -0,0 +1,93 @@
+# Exposing metrics
+
+Date: 2023-05-08
+
+**Status**: Proposed
+
+## Context
+
+Prometheus metrics are a common way to monitor the cluster. Providing metrics
+can be a helpful way to monitor scale sets and health of the ephemeral runners.
+
+## Proposal
+
+There are two main components driving the behaviour of the scale set:
+
+1. ARC controllers responsible for managing kubernetes resources.
+2. The `AutoscalingListener`, driver of the autoscaling solution responsible for
+   describing the desired state.
+
+To expose metrics, we would need to create 3 additional resources:
+
+1. `ServiceMonitor` - resource used by prometheus to match namespaces and
+    services from where it needs to gather metrics
+2. `Service` for the `gha-runner-scale-set-controller` - service that will
+    target ARC controller `Deployment`
+3. `Service` for the `gha-runner-scale-set` listeners - service that will target
+    all listeners for each `AutoscalingRunnerSet`
```

Comment:
I'm thinking that end users will not understand why we have 2 metrics servers for what is 1 controller in their mind. I feel like we're exposing our implementation here.

Is there no way to expose the data from the listener(s) and controller through the same service? Technically it might be challenging and/or not worth the effort, but did we at least consider what it would take?